processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryRunnerFactory.java

```diff
@@ -146,7 +147,9 @@ public SegmentMetadataQueryRunnerFactory(
                           @Override
                           public Sequence<SegmentAnalysis> call() throws Exception
                           {
-                            return input.run(query, responseContext);
```

Comment:
i meant instead of duplicating the code for materializing the result at multiple places, can we reuse the ChainedExecutionQueryRunner here ?